packages/zip-it-and-ship-it/src/runtimes/node/index.ts

```diff
@@ -64,10 +64,19 @@ const zipFunction: ZipFunction = async function ({
     return { config, path: destPath, entryFilename: '' }
   }
 
+  // If the function is inside the plugins modules path, we need to treat that
+  // directory as the base path, not as an extra directory used for module
+  // resolution. So we unset `pluginsModulesPath` for this function.
```

Comment:
This comment is super valuable, and I think it'd be helpful to expand on the "why" in this comment. (My understanding: It's because we expect functions in build plugins to be self-contained/include all their own dependencies; if we don't unset this value, we may resolve missing dependencies using the project's dependencies.)

I don't know if you can make that... uh, less wordy...